Classes/Private/Wrapper/OCTToxAV.m

```diff
@@ -258,7 +388,16 @@ void callstateCallback(ToxAV *cToxAV,
                        uint32_t state,
                        void *userData)
 {
-    // To Do..
+    OCTToxAV *toxAV = (__bridge OCTToxAV *)userData;
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        DDLogCInfo(@"%@: callstateCallback from friend %d with state: %d", cToxAV, friendNumber, state);
+
+
+        if ([toxAV.delegate respondsToSelector:@selector(toxAV:callStateChanged:friendNumber:)]) {
+            [toxAV.delegate toxAV:toxAV callStateChanged:(OCTToxAVCallState)state friendNumber:friendNumber];
```

Comment:
Yeah you are right, not switch. You have to do the following:

```
enum TOXAV_CALL_STATE cState;
OCTToxAVCallState state = 0;

if (cState & TOXAV_CALL_STATE_ERROR) {
    state |= OCTToxAVCallStateError;
}
if (cState & TOXAV_CALL_STATE_FINISHED) {
    state |= OCTToxAVCallStateFinished;
...
}
```

Also please change `uint32_t` to `enum TOXAV_CALL_STATE` for state.